IQSS/dataverse-frontend
# 428 Show New Version Numbers in Publish Dataset Modal

## What this PR does / why we need it:
For Drafts of previously released versions, the Modal will show the new version numbers for Minor and Major Version updates.

This PR also enables "Update Current Version" option, because it the js-dataverse use case has been updated to allow this option

## Which issue(s) this PR closes:

- Closes #428 

## Special notes for your reviewer:

## Suggestions on how to test this:
**Create a test user account in JSF**
1.  Create a new account that is not a super user
2. Login as dataverseAdmin
3. From the Root Collection, choose Edit
4. Give the newly created user "Data Curator" role in the root collection

**Test Publishing from "Data Curator Role"**
1.  From the Root Collection, Create a dataset and publish it using this account
2.  Edit the dataset and choose the publish action again
3. In the publish popup, "Update Current Version" should not be visible. There should be two options:  Minor (1.1) and Major (2.0) 
4. Choose the Minor Option.  After the page is published, the new version should be 1.1
5. Edit the Metadata again, and choose the Publish action.
6. In the Publish Popup, there should be options:  Minor (1.2) and Major (2.0)
7. Choose the Major option. The new version of the Dataset should be 2.0

**Test Update Current Version:**
 Only if you are logged in as a super user, there should be a third Publish Option, "Update Current Version". To test this:
 1. Login as dataverseAdmin
 2. Create a dataset, and publish it
 3. Edit the dataset metadata and choose the Publish Action
 4. Choose the "Update Current Version" option
 5. The new version of the Dataset should have the same version number (1.0)
 

## Does this PR introduce a user interface change? If mockups are available, please link/include them here:

## Is there a release notes update needed for this change?:

## Additional documentation:
